api/woeip/apps/air_quality/serializers.py

```diff
@@ -98,9 +96,10 @@ def create(self, validated_data):
         gps_upload_file = None
         missing_file_errors = []
         for upload_file in upload_files:
-            if "dustrak" in upload_file["file_name"].lower():
+            file_name = upload_file.name
+            if "dustrak" in file_name.lower():
                 dustrak_upload_file = upload_file
-            elif "gps" in upload_file["file_name"].lower():
+            elif "gps" in file_name.lower():
```

Comment:
I would like to revisit the search in the future. I'm still not convinced searching for the name of the file is reliable. Though, this is an Issue (#216 ) for another Pull Request.

Reply:
+1 The file name was meant as a temporary heuristic. It might be a better heuristic to check the file extension. I think it would be even better if the frontend can tell and designate which file is which (e.g. could we add a '`type'` field to each `upload_file`). Edit: just realized this is what #216 describes.

Reply:
(also agreed that this can be addressed in another PR).